File: src/main/java/june_2025/NumOfOneBits.java
package june_2025;

public class NumOfOneBits {

    public static void main(String[] args) {
        int n = 2147483645;

        NumOfOneBits numOfOneBits = new NumOfOneBits();
        int result = numOfOneBits.hammingWeight(n);
        System.out.println(result);

    }

    // O(1) time | O(1) space
    public int hammingWeight(int n) {
        int count = 0;
        while (n > 0) {
            if ((n & 1) == 1) {
                count++;
            }
            n = n >> 1;
        }
        return count;
    }

}

File: python/num_of_one_bits.py
class Solution(object):
    # O(1) time | O(1) space
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        while n > 0:
            if (n & 1) == 1:
                count += 1
            n = n >> 1
        return count
    
solution = Solution()
n = 2147483645
print(solution.hammingWeight(n))    


